firedrake/mg/ufl_utils.py

```diff
@@ -266,10 +266,23 @@ def coarsen_snescontext(context, self, coefficient_mapping=None):
             except CoarseningError:
                 # Assume not something that needs coarsening (e.g. float)
                 new_appctx[k] = v
+
+    _, level = utils.get_level(problem.u_restrict.function_space().mesh())
+    if level == 0:
+        # Use different mat_type on coarsest level
+        opts = PETSc.Options(context.options_prefix)
+        mat_type = opts.getString("mg_coarse_mat_type", default=context.mat_type)
+        pmat_type = context.pmat_type if context.mat_type == mat_type else mat_type
+        pmat_type = opts.getString("mg_coarse_pmat_type", default=pmat_type)
+    else:
+        mat_type = context.mat_type
+        pmat_type = context.pmat_type
```

Comment:
This is really nice to have.
Is there any reason not to do it for levels too?

Reply:
There is only one coarse level. If we add support for `mg_levels_mat_type` then the finest level will always use the mat_type from the problem and the intermidiate levels will use a different one. We cannot easily set up two different relaxations for the the finest level and the intermidiate levels, it is possible but it requires passing prefixes. We could probably support `mg_levels_k_mat_type`, but is this any useful?

Reply:
It seems that if  I add that one here it will probably require new tests and make the PR way more involved. So maybe this point needs more discusion and merits its own PR.

Reply:
I think that's fair enough. The biggest gain is what you already have here, and it looks basically ready to go. I'd be happy with this going through and leaving levels for another PR. Realistically it's a pretty niche case anyway.